client/firecracker_client_factory_test.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"github.com/stretchr/testify/assert"
+	"net"
+	"testing"
+	"time"
+)
+
+func TestNewHttpClientForUNIXSocket(t *testing.T) {
+	done := make(chan bool)
+
+	expectedMessage := "PUT /logger HTTP/1.1\r\nHost: localhost\r\nUser-Agent: Go-http-client/1.1\r\nContent-Length: 0\r\nAccept: application/json\r\nContent-Type: application/json\r\nAccept-Encoding: gzip\r\n\r\n"
+	socketName := "testingUnixSocket.sock"
+	addr, _ := net.ResolveUnixAddr("unix", socketName)
+	listener, _ := net.ListenUnix("unix", addr)
+	defer listener.Close()
+
+	go func() {
+		conn, err := listener.AcceptUnix()
+		if err != nil {
+			t.Error(err)
+		}
+
+		for {
```

Comment:
Does this for loop need to be there? My assumption would be `Read` blocks until something can be read.

Reply:
Indeed! Verified that it's not needed. 
Removed